ldms/src/core/ldms.c

```diff
@@ -2875,6 +2875,276 @@ double ldms_metric_array_get_double(ldms_set_t s, int mid, int idx)
 	return 0;
 }
 
+void ldms_mval_set_char(ldms_mval_t mv, char v)
+{
+	mv->v_char = v;
+}
+
+void ldms_mval_set_u8(ldms_mval_t mv, uint8_t v)
+{
+	mv->v_u8 = v;
+}
+
+void ldms_mval_set_s8(ldms_mval_t mv, int8_t v)
+{
+	mv->v_s8 = v;
+}
+
+void ldms_mval_set_u16(ldms_mval_t mv, uint16_t v)
+{
+	mv->v_u16 = __cpu_to_le16(v);
+}
+
+void ldms_mval_set_s16(ldms_mval_t mv, int16_t v)
+{
+	mv->v_s16 = __cpu_to_le16(v);
+}
+
+void ldms_mval_set_u32(ldms_mval_t mv, uint32_t v)
+{
+	mv->v_u32 = __cpu_to_le32(v);
+}
+
+void ldms_mval_set_s32(ldms_mval_t mv, int32_t v)
+{
+	mv->v_s32 = __cpu_to_le32(v);
+}
+
+void ldms_mval_set_u64(ldms_mval_t mv, uint64_t v)
+{
+	mv->v_u64 = __cpu_to_le64(v);
+}
+
+void ldms_mval_set_s64(ldms_mval_t mv, int64_t v)
+{
+	mv->v_s64 = __cpu_to_le64(v);
+}
+
+void ldms_mval_set_float(ldms_mval_t mv, float v)
+{
+#if LDMS_SETH_F_LCLBYTEORDER == LDMS_SETH_F_LE
```

Comment:
@nichamon, I think __cpu_to_le32 reduces to a "no-op" on LE platforms. Was there an issue with the forced type cast in the __cpu_to_le32? It seems like if there were, then it would break on BE systems for the same reason.

Reply:
No, there was no issue. I have tested it by removing the `#if` condition, and it works fine on my little-endian machine. I think it's safe to remove the `#if` condition. I copied the code from `ldms_metric_set_float()` and pasted it here.

Reply:
What I meant by removing the `#if` condition is to replace 
```
#if LDMS_SETH_F_LCLBYTEORDER ...
     mv->v_f = v;
#else
     *(uint32_t *)&mv->v_f = __cpu_to_le32(*(uint32_t *)&v);
#endif
```
by
```
*(uint32_t *)&mv->v_f = __cpu_to_le32(*(uint32_t *)&v);
```